test/git_checkout_test.go

```diff
@@ -53,7 +53,61 @@ func TestGitPipelineRun(t *testing.T) {
 			defer tearDown(t, c, namespace)
 
 			t.Logf("Creating Git PipelineResource %s", gitSourceResourceName)
-			if _, err := c.PipelineResourceClient.Create(getGitPipelineResource(namespace, revision, "true", "", "", "")); err != nil {
+			if _, err := c.PipelineResourceClient.Create(getGitPipelineResource(namespace, revision, "", "true", "", "", "")); err != nil {
+				t.Fatalf("Failed to create Pipeline Resource `%s`: %s", gitSourceResourceName, err)
+			}
+
+			t.Logf("Creating Task %s", gitTestTaskName)
+			if _, err := c.TaskClient.Create(getGitCheckTask(namespace)); err != nil {
+				t.Fatalf("Failed to create Task `%s`: %s", gitTestTaskName, err)
+			}
+
+			t.Logf("Creating Pipeline %s", gitTestPipelineName)
+			if _, err := c.PipelineClient.Create(getGitCheckPipeline(namespace)); err != nil {
+				t.Fatalf("Failed to create Pipeline `%s`: %s", gitTestPipelineName, err)
+			}
+
+			t.Logf("Creating PipelineRun %s", gitTestPipelineRunName)
+			if _, err := c.PipelineRunClient.Create(getGitCheckPipelineRun(namespace)); err != nil {
+				t.Fatalf("Failed to create Pipeline `%s`: %s", gitTestPipelineRunName, err)
+			}
+
+			if err := WaitForPipelineRunState(c, gitTestPipelineRunName, timeout, PipelineRunSucceed(gitTestPipelineRunName), "PipelineRunCompleted"); err != nil {
+				t.Errorf("Error waiting for PipelineRun %s to finish: %s", gitTestPipelineRunName, err)
+				t.Fatalf("PipelineRun execution failed")
+			}
+		})
+	}
+}
+
+// Test revision fetching with refspec specified
+func TestGitPipelineRunWithRefspec(t *testing.T) {
+	t.Parallel()
+
+	// Pair of revision, refspec
+	var pairs = []string{
```

Comment:
The approach we take elsewhere in our testing is to build a table of test cases with descriptions, inputs, and expected outputs (if any).

For example, I'd approach this test as follows:

```go
for _, tc := range []struct{
	description string
	revision string
	refspec string
}{{
	description: "Fetch refs/tags/v0.1.0 alongside master and checkout the master branch",
	revision: "master",
	refspec: "refs/tags/v0.1.0:refs/tags/v0.1.0 refs/heads/master:refs/heads/master",
}, {
	description: "Checkout specific revision from refs/pull/1009/head's commit chain",
	revision: ""968d5d37a61bfb85426c885dc1090c1cc4b33436"",
	refspec: "refs/pull/1009/head",
}, {
	// ...and so on...
}} {
	t.Run(tc.description, func(t *testing.T) {
		// test body
	})
}
```

The benefit is that this moves the comments into test names, which get logged on failure, and it adds a clean field name for each of the entries in your current pairs list.  It would also match more closely what we're doing elsewhere, which is nice from a pure code-style perspective.

Reply:
@sbwsg Thanks for the explanation. That's a great way to organize the test cases. I've updated the PR.